core/corehttp/gateway_handler.go

```diff
@@ -766,8 +766,8 @@ func (i *gatewayHandler) buildIpfsRootsHeader(contentPath string, r *http.Reques
 		the last root (responsible for specific article) may not change at all.
 	*/
 	var sp strings.Builder
-	var pathRoots []string
 	pathSegments := strings.Split(contentPath[6:], "/")
+	pathRoots := make([]string, 0, len(pathSegments))
```

Comment:
Actually here it doesn't make sense to append as this append is conditional:
```go
if root == "" {
  continue
}
```
In thoses cases, trust append.
```suggestion
	var pathRoots []string
	pathSegments := strings.Split(contentPath[6:], "/")
```